packages/ethpm-v1/lib/ethpm-v1.js

```diff
@@ -1,16 +1,76 @@
 const expect = require("@truffle/expect");
 const TruffleError = require("@truffle/error");
-const Networks = require("./networks");
+const Networks = require("@truffle/core/lib/networks");
 const EthPM = require("ethpm");
 const EthPMRegistry = require("ethpm-registry");
 const Web3 = require("web3");
+const registryAbi = require("./registryAbi");
 const { createInterfaceAdapter } = require("@truffle/interface-adapter");
 const path = require("path");
 const fs = require("fs");
 const OS = require("os");
 
-const Package = {
-  install: async function(options, callback) {
+const PackageV1 = {
+  packages: async options => {
+    options.logger.log(
+      `ethpmV1 is deprecated. Please consider updating your packages to ethpmV3.`
+    );
+    try {
+      expect.options(options, [
+        "ethpm",
+        "logger",
+        "working_directory",
+        "contracts_build_directory",
+        "networks"
+      ]);
+    } catch (err) {
+      throw new TruffleError(
+        `Invalid ethpm configuration in truffle-config: ${err.message}`
+      );
+    }
+
+    const ethpmV1Network = "ropsten";
+    if (!options.networks[ethpmV1Network]) {
+      throw new TruffleError(
+        `Please include a provider in your networks config for the Ropsten testnet, the testnet where the ethpmV1 registry is deployed.`
+      );
+    }
+    const provider = options.networks["ropsten"].provider();
```

Comment:
Will this always be a method? I feel like this property is usually accessed like `options.networks[<name>].provider`. I could be wrong, I'll have to check this out.